src/EnergyPlus/DataDaylighting.hh

```diff
@@ -95,88 +96,89 @@ namespace DataDaylighting {
         DElightDaylighting
     };
 
-    // Parameters for "Lighting Control Type"
-    enum class iLtgCtrlType
+    // Parameters for "Lighting Control Type" - these are the values expected by DElight
+    enum class LtgCtrlType
     {
+        Invalid = 0,
         Continuous = 1,
         Stepped = 2,
-        ContinuousOff = 3
+        ContinuousOff = 3,
     };
 
-    struct IntWinAdjZoneExtWinStruct // nested structure for ZoneDaylight
+    constexpr std::array<std::string_view, 4> LtgCtrlTypeNamesUC = {"INVALID", "CONTINUOUS", "STEPPED", "CONTINUOUSOFF"};
+
+    struct IntWinAdjEnclExtWinStruct // nested structure for EnclDaylightCalc
     {
         // Members
-        int SurfNum;           // exterior window index
-        int NumOfIntWindows;   // count of interior windows associated with this ext win
-        Array1D_int IntWinNum; // index numbers for interior windows assoc with this ext win
-
-        // Default Constructor
-        IntWinAdjZoneExtWinStruct() : SurfNum(0), NumOfIntWindows(0)
-        {
-        }
+        int SurfNum = 0;         // exterior window index
+        int NumOfIntWindows = 0; // count of interior windows associated with this ext win
+        Array1D_int IntWinNum;   // index numbers for interior windows assoc with this ext win
```

Comment:
I see this a lot. Is a separate `NumOfIntWindows` variable necessary? Why not use `IntWinNums.size()`?

Reply:
> I see this a lot. Is a separate NumOfIntWindows variable necessary? Why not use IntWinNums.size()?

Valid question. Some of these were already there and I followed the same pattern, sometimes, I got frustrated by the compiler complaining about types when using the size as a for loop limit, and in some cases, the actual number of things can be less than the allocated size. Not great, but it's the way some arrays are at the moment.